component/componentstatus/instance.go

```diff
@@ -3,14 +3,89 @@
 
 package componentstatus // import "go.opentelemetry.io/collector/component/componentstatus"
 
-import "go.opentelemetry.io/collector/component"
+import (
+	"slices"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+)
 
 // InstanceID uniquely identifies a component instance
 //
 // TODO: consider moving this struct to a new package/module like `extension/statuswatcher`
```

Comment:
This TODO needs moved back down to InstanceID (or removed, I think it might be a bad copy and paste from `StatusWatcher`.